treemap.swt/src/main/java/de/engehausen/treemap/swt/TreeMap.java

```diff
@@ -57,6 +57,13 @@
 	protected BuildControl buildControl;
 	protected Image image;
 
+	/**
+	 * Indicates whether the mouse cursor should be changed to {@link SWT#CURSOR_WAIT} during recalculation.
+	 *
+	 * @author Mario Marinato
+	 */
+	private boolean changeCursorOnRecalculate = true;

```

Comment:
As a general rule, I do not initialize member variables outside of constructors of classes (a style question - I see I violated that with initializing a default renderer, I should have done that only in the main constructor). So the member can be declared here (make it protected for subclasses to be able to access it): `protected boolean changeCursorOnRecalculate;` and create a new main constructor:

```java
public TreeMap(final Composite composite, final boolean supportNavigation, final boolean changeCursorOnRecalculate) {
  ...
  this.changeCursorOnRecalculate = changeCursorOnRecalculate;
}
```

The previous main constructor needs to change:

```java
public TreeMap(final Composite composite, final boolean supportNavigation) {
  this(composite, supportNavigation, true);
}
```

It would make sense to change the single argument constructor to call the main constructor immediately:

```java
public TreeMap(final Composite composite) {
  this(composite, true, true);
}
```

The same change to the SWT version of TreeMap would be required to the Swing variant too (add the cursor flag).

Reply:
Makes perfect sense.  I'll change that.